docs/contracts.rst

```diff
@@ -284,8 +284,38 @@ It will generate a function of the following form::
         b = data[arg1][arg2][arg3].b;
     }
 
-Note that the mapping in the struct is omitted because there
-is no good way to provide the key for the mapping.
+    .. note::
+        The mapping in the struct is omitted because there is no good way to provide the key for the mapping.
+
+If you have a `public` state variable of array type, then it only returns
```

Comment:
`it` is perhaps a bit ambiguous here. What about `then you can only retrieve single elements of the array via the generated getter function. This mechanism exists to limit the gas costs of returning an entire array`

Reply:
Good call